### PR TITLE
wifi: Add nRF7001DK support to lock sample

### PR DESCRIPTION
**Pull request by krish2718**

### CODE REVIEW

File: samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+mcuboot:
```
Comment (Damian-Nordic) [inline samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml:1]: This clashes with #14281 but I'll let you and @ArekBalysNordic decide which PR goes first :)

**Review by sachinthegreen — APPROVED** [review]

File: samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml (hunk 1 above)
Comment (krish2718) [reply samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml:1]: @ArekBalysNordic WDYT? Shall I go ahead with this merge?

Comment (ArekBalysNordic) [reply samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml:1]: We changed the layout recently. Please rename it to `pm_static_nrf7002dk_nrf7001_nrf5340_cpuapp.yml` and move it to the main sample's directory. There is no "configuration" directory within the sample anymore on NCS main.

Comment (ArekBalysNordic) [reply samples/matter/lock/configuration/nrf7002dk_nrf7001_nrf5340_cpuapp/pm_static_dfu.yml:1]: Do we want to support the "release" version? If so you can create the same file but with the suffix `_release.yml`: `pm_static_nrf7002dk_nrf7001_nrf5340_cpuapp_release.yml` and store it in the main sample's directory.

**Review by ArekBalysNordic — CHANGES_REQUESTED** [review]
What about mentioning that change in the release notes and README.rst file?

**Comment by krish2718** [comment]
> What about mentioning that change in the release notes and README.rst file?

Not sure,I don't think it warrants a release note update? and README should automatically updated `table-from-sample.yaml`

**Review by ArekBalysNordic — APPROVED** [review]
Thank you for the .yaml files alignment. Regarding docs, the mentioned table won't be updated because you didn't create an additional target in the samples.yaml file for nrf7001. The question is, do we want to support this target @LuDuda? If so we need to update docs, otherwise we can go ahead with it, as it is, so approving. We can add the docs later as well, since it is the main branch.

File: samples/matter/lock/boards/nrf7002dk_nrf7001_nrf5340_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/ {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
+	};
+};
+
+/* Set IPC thread priority to the highest value to not collide with other threads. */
+&ipc0 {
+    zephyr,priority = <0 PRIO_COOP>;
```
Comment (nordicjm) [inline samples/matter/lock/boards/nrf7002dk_nrf7001_nrf5340_cpuapp.overlay:17]: tabs for overlay files

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -254,10 +254,13 @@ Matter samples
...
-* Added test event triggers to all Matter samples.
-  By utilizing the test event triggers, you can simulate various operational conditions and responses in your Matter device without the need for external setup.
+* Added:
+
+   * Test event triggers to all Matter samples.
+     By utilizing the test event triggers, you can simulate various operational conditions and responses in your Matter device without the need for external setup.
 
-  To get started with using test event triggers in your Matter samples and to understand the capabilities of this feature, refer to the :ref:`ug_matter_test_event_triggers` page.
+     To get started with using test event triggers in your Matter samples and to understand the capabilities of this feature, refer to the :ref:`ug_matter_test_event_triggers` page.
+   * Support for emulation of nRF7001 Wi-Fi companion IC on nrf7002 DK to Matter Lock Sample.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:263]:
```suggestion

* :ref:`matter_lock_sample` sample:

   * Added support for emulation of the nRF7001 Wi-Fi companion IC on the nRF7002 DK.
```
So far, we've had changes specific to a single sample under an entry for that sample.

File: samples/matter/lock/README.rst [hunk 4]
```diff
@@ -44,7 +44,7 @@ IPv6 network support
 The development kits for this sample offer the following IPv6 network support for Matter:
 
 * Matter over Thread is supported for ``nrf52840dk_nrf52840``, ``nrf5340dk_nrf5340_cpuapp``, and ``nrf21540dk_nrf52840``.
-* Matter over Wi-Fi is supported for ``nrf5340dk_nrf5340_cpuapp`` with the ``nrf7002ek`` shield attached or for ``nrf7002dk_nrf5340_cpuapp``.
+* Matter over Wi-Fi is supported for ``nrf5340dk_nrf5340_cpuapp`` with the ``nrf7002ek`` shield attached (2.4 GHz and 5 GHz), for ``nrf7002dk_nrf5340_cpuapp`` (2.4 GHz and 5 GHz) or for ``nrf7002dk_nrf7001_nrf5340_cpuapp`` (2.4 GHz only).
```
Comment (wiba-nordic) [inline samples/matter/lock/README.rst:47]:
```suggestion
* Matter over Wi-Fi is supported for ``nrf5340dk_nrf5340_cpuapp`` with the ``nrf7002ek`` shield attached (2.4 GHz and 5 GHz), for ``nrf7002dk_nrf5340_cpuapp`` (2.4 GHz and 5 GHz), or for ``nrf7002dk_nrf7001_nrf5340_cpuapp`` (2.4 GHz only).
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -160,9 +160,6 @@ nRF Desktop
     You must use the alias to specify the QDEC instance used for scroll wheel, if your board supports multiple QDEC instances (for example ``nrf54l15pdk_nrf54l15_cpuapp``).
     You do not need to define the alias if your board supports only one QDEC instance, because in that case, the wheel module can rely on the ``qdec`` DT label provided by the board.
 
-* Updated the :kconfig:option:`CONFIG_BT_ADV_PROV_TX_POWER_CORRECTION_VAL` Kconfig option value in the nRF52840 Gaming Mouse configurations with the Fast Pair support.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:163]: Why is this line getting removed? Merge conflict?

Comment (ArekBalysNordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:163]: Oh, that's weird... There was no merge conflict at all, so Git removed it after the rebase.

**Review by wiba-nordic — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]
sample.yaml file changes OK

**Pull request merged**